src/main/java/org/embulk/input/zendesk/stream/PagingSpliterator.java

```diff
@@ -0,0 +1,129 @@
+package org.embulk.input.zendesk.stream;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.http.HttpStatus;
+import org.embulk.config.ConfigException;
+import org.embulk.input.zendesk.ZendeskInputPlugin;
+import org.embulk.input.zendesk.clients.ZendeskRestClient;
+import org.embulk.input.zendesk.models.Target;
+import org.embulk.input.zendesk.models.ZendeskException;
+import org.embulk.input.zendesk.utils.ZendeskUtils;
+
+import java.util.Iterator;
+import java.util.Spliterator;
+import java.util.function.Consumer;
+
+public abstract class PagingSpliterator<E> implements Spliterator<E>
```

Comment:
I feel we're building an abstract parent class which contains all of common methods for its child classes, so it behaves pretty much similar as utility class. What if more new targets with different endpoints will be used? Does this abstract need to be modified for those new endpoint?

I'd suggest restructure the `PagingSpliterator` and its child classes to bring clear sense.

Reply:
fixed